packages/wdio-allure-reporter/src/reporter.ts

```diff
@@ -1,82 +1,313 @@
+import {
+    TYPE as DescriptionType
 } from './types.js'
 
-const require = createRequire(import.meta.url)
-
-/**
- * Allure v1 has no proper TS support
- * ToDo(Christian): update to Allure v2 (https://github.com/webdriverio/webdriverio/issues/6313)
- */
-const Allure = require('allure-js-commons')
-const Step = require('allure-js-commons/beans/step.js')
-
 export default class AllureReporter extends WDIOReporter {
-    private _allure: any
+    private _allure: AllureRuntime
     private _capabilities: Capabilities.RemoteCapability
     private _isMultiremote?: boolean
     private _config?: Options.Testrunner
-    private _lastScreenshot?: string
     private _options: AllureReporterOptions
     private _consoleOutput: string
     private _originalStdoutWrite: Function
-    private _addConsoleLogs: boolean
-    private _startedFeatures: SuiteStats[] = []
+
+    // TODO: i guess better to rename the property to make it easy to understand that we actually work with stack
+    _runningUnits: Array<AllureGroup | AllureTest | AllureStep> = []
 
     constructor(options: AllureReporterOptions = {}) {
-        const outputDir = options.outputDir || 'allure-results'
+        const { outputDir = 'allure-results', ...rest } = options
+
         super({
-            ...options,
+            ...rest,
             outputDir,
         })
-        this._addConsoleLogs = false
         this._consoleOutput = ''
         this._originalStdoutWrite = process.stdout.write.bind(process.stdout)
-        this._allure = new Allure()
+        this._allure = new AllureRuntime({
+            resultsDir: outputDir,
+        })
         this._capabilities = {}
         this._options = options
 
-        this._allure.setOptions({ targetDir: outputDir })
         this.registerListeners()
 
-        this._lastScreenshot = undefined
-
         const processObj:any = process
-        if (options.addConsoleLogs || this._addConsoleLogs) {
+
+        if (options.addConsoleLogs) {
             processObj.stdout.write = (chunk: string, encoding: BufferEncoding, callback:  ((err?: Error) => void)) => {
                 if (typeof chunk === 'string' && !chunk.includes('mwebdriver')) {
                     this._consoleOutput += chunk
                 }
+
                 return this._originalStdoutWrite(chunk, encoding, callback)
             }
         }
     }
 
+    get currentSuite(): AllureGroup | undefined {
+        return findLast(this._runningUnits, (unit) => unit instanceof AllureGroup) as AllureGroup | undefined
+    }
+
+    get currentTest(): AllureTest | undefined {
+        return findLast(this._runningUnits, (unit) => unit instanceof AllureTest) as AllureTest | undefined
+    }
+
+    get currentStep(): AllureStep | undefined {
+        return findLast(this._runningUnits, (unit) => unit instanceof AllureStep) as AllureStep | undefined
+    }
+
+    get currentAllureSpec(): AllureTest | AllureStep | undefined {
```

Comment:
The name is a bit confusing. It's probably a good idea to name it as `currentAllureTestOrStep` to reflect what it actually represents.

Reply:
Agree, :+1:

Reply:
:+1: